lib/sendgrid/helpers/mail/mail.rb

```diff
@@ -82,33 +78,17 @@ def add_custom_arg(custom_arg)
     end
 
     def asm=(asm)
-      @asm = asm
-    end
-
-    def asm
       @asm.nil? ? nil : @asm.to_json
     end
```

Comment:
This change is causing test failures, should be `def asm`, since we want a custom reader, not writer.

Reply:
I think that a reverse merge (merge main into issue) caused some confusion, because I did't made this change.
When I created the Pull Request it was based on master, which does not exist anymore.
So I think that the best to do is to close this PR now.

Reply:
Ah, I [see the change](https://github.com/sendgrid/sendgrid-ruby/pull/338/commits/9d20e856bd718ecdc579730daccce2a4ea8a6ae7). The default branch name was recently renamed and a script was run to update existing PRs. Most of this looks good, so I'm inclined to revert the commit that broke your code and re-review.